src/main/redux/sagas/i18n.ts

```diff
@@ -7,23 +7,32 @@
 
 import { LocaleConfigIdentifier, LocaleConfigValueType } from "readium-desktop/common/config";
 import { i18nActions } from "readium-desktop/common/redux/actions";
-import { takeTyped } from "readium-desktop/common/redux/typed-saga";
 import { ConfigRepository } from "readium-desktop/main/db/repository/config";
 import { diMainGet } from "readium-desktop/main/di";
-import { all, call } from "redux-saga/effects";
+import { all, call, takeEvery } from "redux-saga/effects";
 
-export function* localeWatcher() {
-    while (true) {
-        const action = yield* takeTyped(i18nActions.setLocale.build);
-        const translator = diMainGet("translator");
-        translator.setLocale(action.payload.locale);
+function* setLocale(action: i18nActions.setLocale.TAction) {
+    const translator = diMainGet("translator");
+    const configRepository: ConfigRepository<LocaleConfigValueType> = diMainGet("config-repository");
 
-        const configRepository: ConfigRepository<LocaleConfigValueType> = diMainGet("config-repository");
-        yield call(() => configRepository.save({
+    const configRepositorySave = () =>
+        configRepository.save({
             identifier: LocaleConfigIdentifier,
-            value: { locale: action.payload.locale },
-        }));
-    }
+            value: {
+                locale: action.payload.locale,
+            },
+        });
+    const translatorSetLocale = () =>
+        translator.setLocale(action.payload.locale);
```

Comment:
Missing `await` when invoking a Promise-returning function (this would have been flagged-up if the TypeScript project was configured with stricter checks).
PS: `async` must be added in `    const translatorSetLocale = async () =>` too.
PS: Redux `call` will automatically handle the `Promise<void>`-returning `translatorSetLocale` function.

Same remark with `configRepository.save()` (above).

Reply:
https://github.com/readium/readium-desktop/pull/907/commits/73dbaf736242aa558ac43cc7b35d05960b1c9af6